assets/component-facets.css

```diff
@@ -331,18 +336,111 @@
 .facets-layout-grid .facets__label {
   display: flex;
   flex-direction: column;
-  gap: 0.8rem;
   padding: 0;
   height: 100%;
   font-size: 1.3rem;
 }
 
-.facets__item label,
-.facets__item input[type='checkbox'] {
+/* Image filter specific styles */
+.facets-layout-grid .facets__image-wrapper {
+  display: block;
+  position: relative;
+  aspect-ratio: 1 / 1;
+  min-height: 0;
+  padding: 0.4rem;
+}
+
+.facets__image {
+  width: 100%;
+  height: 100%;
+  object-fit: contain;
+}
+
+/* Default state */
+.facets-layout-grid--image .facets__label {
+  outline-style: solid;
+  outline-color: transparent;
+  transition-property: outline-color, outline-width, box-shadow;
+  transition-duration: var(--duration-short);
+  transition-timing-function: ease;
+}
+
+.facets-layout-grid--image .facet-checkbox__text {
+  padding: 0.4rem;
+}
+
+/* Active state */
+.facets-layout-grid--image .facets__label.active {
+  outline-color: rgb(var(--color-foreground));
+  outline-width: 0.1rem;
+}
+
+/* Hover state */
+.facets-layout-grid--image .facets__label:hover {
+  outline-color: rgba(var(--color-foreground), 0.4);
+  outline-width: 0.2rem;
+}
+
+/* Focus visible */
+.facets-layout-grid--image .facets__label:has(:focus-visible) {
+  outline-color: rgba(var(--color-foreground), 0.5);
+  outline-width: 0.2rem;
+  box-shadow: 0px 0px 3px 1px rgba(var(--color-foreground), 0.25);
+}
+
+/* Focused state */
+.facets-layout-grid--image .facets__label.active:has(:focus-visible) {
+  outline-color: rgb(var(--color-foreground));
+  outline-width: 0.1rem;
+  box-shadow: 0 0 0 0.3rem rgb(var(--color-background)), 0 0 0 0.5rem rgba(var(--color-foreground), 0.5),
+    0 0 0.7rem 0.1rem rgba(var(--color-foreground), 0.25);
+}
+
+/* Disabled state */
+.facets-layout-grid--image .facets__label.disabled {
+  /* Note: disabled gets its transparency set by a parent, so no need to set an alpha here. Also, we move the outline to the child when disabled */
+  outline: none;
+}
+
+.facets-layout-grid--image .facets__label.disabled .facets__image-wrapper {
+  outline: 0.1rem solid rgb(var(--color-foreground));
+}
+
+.facets-layout-grid--image .facets__label.disabled .facets__image-wrapper:before {
+  content: '';
+  position: absolute;
+  bottom: 0;
+  left: 0;
+  /* 100% * square root of 2, this is not a magic number, it is math! */
+  width: 141.4%;
+  height: 0.1rem;
```

Comment:
The widths of the cross out line between an image and a swatch are noticeably different. Is this intentional?  The `image` on is much thinner

Image is calculated by `height` of the div

![](https://screenshot.click/13-08-ujibb-9ts43.png)

Swatch is calculated by `width` of the pseudo elements

![](https://screenshot.click/13-09-mq7xd-zmniw.png)